apps/chat/src/components/Chat/ChatMessage/MessageSchema/AssistantSchema.tsx

```diff
@@ -0,0 +1,75 @@
+import { memo, useCallback, useMemo } from 'react';
+
+import { ChatActions } from '@/src/store/chat/chat.reducer';
+import { ConversationsSelectors } from '@/src/store/conversations/conversations.reducers';
+import { useAppDispatch, useAppSelector } from '@/src/store/hooks';
+
+import { FormSchema } from '@/src/components/Chat/ChatMessage/MessageSchema/FormSchema';
+
+import { Message, MessageFormValueType } from '@epam/ai-dial-shared';
+
+interface AssistantSchemaProps {
+  message: Message;
+  isLastMessage: boolean;
+}
+
+export const AssistantSchema = memo(function AssistantSchema({
+  message,
+  isLastMessage,
+}: AssistantSchemaProps) {
+  const dispatch = useAppDispatch();
+
+  const isPlayback = useAppSelector(
+    ConversationsSelectors.selectIsPlaybackSelectedConversations,
+  );
+
+  const schema = useMemo(() => message.custom_content?.form_schema, [message]);
+
+  const descriptions = useMemo(() => {
+    if (!schema) return [];
+
+    return Object.values(schema.properties)
+      .map(({ description }) => description)
+      .filter(Boolean);
+  }, [schema]);
+
+  const handleChange = useCallback(
+    (property: string, value: MessageFormValueType, submit?: boolean) => {
+      if (schema) {
+        const populateText = schema.properties[property]?.oneOf?.find(
+          (option) => option.const === value,
+        )?.['dial:widgetOptions']?.populateText;
+
+        dispatch(
+          ChatActions.setFormValue({
+            property,
+            content: populateText,
+            value,
+            submit,
+          }),
+        );
+      }
+    },
+    [dispatch, schema],
+  );
+
+  if (!schema) return null;
```

Comment:
to avoid several hooks and other stuff will work each time even no schema it would be better to create wrapper like
```
export const AssistantSchema = memo(function AssistantSchema(props: AssistantSchemaProps) {
   if (!props.message.custom_content?.form_schema) return null;
   return <AssistantSchemaView {...props}>
}
```